api/features/versioning/versioning_service.py

```diff
@@ -101,6 +105,308 @@ def get_current_live_environment_feature_version(
     )
 
 
+def update_flag(
```

Comment:
These functions perform multiple database writes without `@transaction.atomic`. Can we wrap them?

Reply:
We don't have any early exits or conditional paths here that could leave things in a bad state - it's just a straight line of create → update → save. If any step fails, it raises and stops.

  Also, if something does fail midway, we'd actually want to know about it. A 500 surfacing is useful signal - silently rolling back could hide bugs we should be investigating.

  Worth noting that FeatureState is one of our hottest tables, so holding transaction locks longer than needed would hurt concurrent traffic.

  Happy to add one if there's a specific consistency case I'm missing though!